gwpy/timeseries/timeseries.py

```diff
@@ -1388,28 +1388,132 @@ def demodulate(self, f, stride=1, exp=False, deg=True):
         >>> ax = plot.gca()
         >>> ax.set_ylabel('Strain Amplitude at 331.3 Hz')
         >>> plot.show()
+
+        See also
+        --------
+        TimeSeries.heterodyne
+            for the underlying heterodyne detection method
         """
+        # stride through the TimeSeries and heterodyne at a single frequency
+        phase = (2 * numpy.pi * f *
+                 self.dt.astype('float32').decompose().value *
```

Comment:
@mattpitkin, I was wrong before, apologies about that, can you remove the the conversion to `float32`?

```suggestion
                 self.dt.decompose().value *
```

**Context:** casting to `float32` causes a loss of precision, which breaks the unit tests for `TimeSeries.demodulate`. More to the point, the loss of precision is unacceptable since it leads to a systematic error in phase at the level of ~1 degree, which is on the same order as the current calibration uncertainty for advanced LIGO.

I was able to profile this method before and after your changes; before the change, demodulating 12 hours of 4 kHz data takes about 4.5 GB, while the same operation after your changes takes about 6.5 GB. I believe this is an acceptable trade-off to get the precision necessary to demodulate advanced LIGO's calibration lines, which is a mission-critical downstream use of this method.